### ISSUE TITLE
Implement rolling_exp for dask arrays

### ISSUE THREAD

**Issue by heerad** [pydata/xarray#4474]
**Is your feature request related to a problem? Please describe.**
I use dask-based chunking on my arrays regularly and would like to leverage the efficient `numbagg` implementation of `move_exp_nanmean()` with `rolling_exp()`.

**Describe the solution you'd like**
It's possible to compute a rolling exp mean as a function of rolling exp means of contiguous, non-overlapping subsets (chunks). You just need to first "un-normalize" the rolling_exps of each chunk in order to split them into their corresponding numerators and denominators (see the `ewm` definition [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.ewm.html) under `adjust=True`). The normalization factor (denominator) to multiply back in to the chunk's `move_exp_nanmean()` in order to un-normalize it (numerator) is just the `move_exp_nanmean()` of 1's, replaced with NA's wherever the underlying data was also NA.

Then, scale each chunk's numerator and denominator series (derived from their `move_exp_nanmean()` series via above) down according to how many "lags-ago" they were, sum the rescaled numerators and denominators across chunks, and finally divide the total summed numerators and denominators.

**Describe alternatives you've considered**
I implemented my own inefficient weighted rolling mean using xarray's `rolling()`. This requires a bunch of duplicate computation as the window gets shifted.

**Comment by max-sixty** [comment]
I think we'd be open to this

**Comment by heerad** [comment]
Great! This will be a common use-case for me, and I imagine others who are doing any sort of time series computation on large datasets.

**Comment by max-sixty** [comment]
I have less experience with dask than others — are simple rolling means already implemented across chunks? That would be a much easier place to start...

**Comment by heerad** [comment]
Yes, see http://xarray.pydata.org/en/stable/computation.html#rolling-window-operations.

`rolling` works with dask, but `rolling_exp` does not.

**Comment by max-sixty** [comment]
OK, I don't see references there to dask, but thanks for letting me know. Do you know where that code is?

**Comment by heerad** [comment]
Looks like it's all in here: https://github.com/pydata/xarray/blob/6d8ac11ca0a785a6fe176eeca9b735c321a35527/xarray/core/dask_array_ops.py

**Comment by max-sixty** [comment]
Thanks @heerad . Looks like we're doing it with overlaps rather than any rolling-specific algo.

Agree that approach won't work with exp. OTOH your suggestion could be generalized to improve performance in all rolling operations, depending on the relative costs of sending the underlying overlapping data vs. sending only the edges of the numerator / denominator and doing another pass.

If anyone has experience with similar operations, feel free to offer suggestions.
